jsonschema-use-cases.xml

```diff
@@ -0,0 +1,300 @@
+            </section>
+
+            <section title="Domain-specific language">
+                <t>
+                    Application authors may use an entirely self-contained inside an application that consumes JSON. By using a declarative language, the application requirements can be better optimized by the application.
+                </t>
+                <t>
+                    Application authors may define custom hooks and processing for the JSON (without need for standardizing the customization).
+                </t>
+                <t>
+                    The only interoperability consideration here is that updates to the validator library must not change the set of valid JSON documents, unless the developer specifically opts into such a breaking change (e.g. by upgrading the library to a new major version number).
+                </t>
+            </section>
+
+            <section title="A common vocabulary">
+                <t>
+                    A team of developers write two similar applications, for different platforms, in different languages. The application downloads and reads JSON documents. They want to make sure that both applications accept or reject JSON with identical behavior, so they write a single JSON Schema and deploy it to both applications.
+                </t>
+                <t>
+                    The only interoperability consideration here is that the two applications, given the same schema, must both be reasonably expected to support the same behavior and operate in the same manner.
+                </t>
+            </section>
+
+            <section title="Model-Driven UI constraints">
+                <t>
+                    When a server declares constraints that a submission must meet, there is a need for the user interface to receive these constraints to provide model-driven validation of permissible values, making the form more accessible to the user.
+                </t>
+                <t>
+                    If the schema is ambiguous in any way, or is up to the discretion of the customer's user-agent, some customers may have a difficult time submitting a correct request.
+                </t>
+            </section>
+
+            <section title="Fuzzing, enumeration, and generation">
+                <t>
+                    Security applications need to generate examples of JSON documents within the valid set, and outside the valid set.
+                </t>
+            </section>
+
+            <section title="Embedded database constraints">
+                <t>
+                    A database that uses JSON may need a way to declare, enforce, or guarantee certain constraints on the JSON document being stored. The database may use JSON Schema as a way to annotate certain fields as having a special meaning for uniqueness or indexing purposes.
+                </t>
+            </section>
+
+            <section title="Partial validation" anchor="partial-validation">
```

Comment:
This section seems like it would be better covered as an "applicability" section of the spec.